src/main/java/com/artipie/scripting/Script.java

```diff
@@ -0,0 +1,251 @@
+
+        /**
+         * Ctor.
+         * @param type Name of scripting engine.
+         * @param script Script code
+         */
+        public StandardScript(final ScriptType type, final String script) {
+            this.engine = Script.MANAGER.getEngineByName(type.toString());
+            this.script = Objects.requireNonNull(script, "Script is null");
+        }
+
+        @Override
+        public Result call() throws ScriptException {
+            final Result result = new Result();
+            result.value(this.engine.eval(this.script, result.context()));
+            return result;
+        }
+
+        @Override
+        public Result call(final Map<String, Object> vars) throws ScriptException {
+            final Result result = new Result(vars);
+            result.value(this.engine.eval(this.script, result.context()));
+            return result;
+        }
+    }
+
+    /**
+     * Precompiled implementation of {@link Script}.
+     * Should be used for multiple invocations of script.
+     * @since 0.30
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    class PrecompiledScript implements Script {
+        /**
+         * Compiled script.
+         */
+        private final CompiledScript script;
+
+        /**
+         * Ctor.
+         * @param type Name of scripting engine.
+         * @param script Script code
+         */
+        public PrecompiledScript(final ScriptType type, final String script) {
+            final ScriptEngine engine = Script.MANAGER.getEngineByName(type.toString());
+            if (!(engine instanceof Compilable)) {
+                throw new ArtipieException(
+                    String.format("Scripting engine '%s' does not support compilation", engine)
+                );
+            }
+            try {
+                this.script = ((Compilable) engine).compile(script);
+            } catch (final ScriptException exc) {
+                throw new ArtipieException(exc);
+            }
+        }
+
+        @Override
+        public Result call() throws ScriptException {
+            final Result result = new Result();
+            result.value(this.script.eval(result.context()));
+            return result;
+        }
+
+        @Override
+        public Result call(final Map<String, Object> vars) throws ScriptException {
+            final Result result = new Result(vars);
+            result.value(this.script.eval(result.context()));
+            return result;
+        }
+    }
+
+    /**
+     * Result of script invocation.
+     * @since 0.30
+     */
+    @SuppressWarnings(
+        {
+            "PMD.OnlyOneConstructorShouldDoInitialization",
+            "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
+        }
+    )
+    class Result {
+        /**
+         * Script context.
+         */
+        private final ScriptContext context;
+
+        /**
+         * Resulting value.
+         */
+        private Object val;
```

Comment:
Is this field used anywhere? The setter is private and not called inside this class

Reply:
I updated comments for the object - it is required to get result value from `.eval()`. 
For `VertxMain.stop()` - should I remove whole method? I don't see its usage too.

Reply:
I think yes, it would be better to remove it, so it do not cause any misunderstanding